pysteps/feature/tstorm.py

```diff
@@ -0,0 +1,320 @@
+"""
+pysteps.feature.tstorm
+======================
+
+Thunderstorm cell detection module, part of Thunderstorm Detection and Tracking (DATing)
+This module was implemented following the procedures used in the TRT Thunderstorms
+Radar Tracking algorithm (:cite:`TRT2004`) used operationally at MeteoSwiss.
+Modifications include advecting the identified thunderstorms with the optical flow
+obtained from pysteps, as well as additional options in the thresholding.
+
+References
+...............
+:cite:`TRT2004`
+
+@author: mfeldman
+
+.. autosummary::
+    :toctree: ../generated/
+
+    detection
+    belonging
+    longdistance
+    get_profile
+"""
+
+import numpy as np
+import scipy.ndimage as ndi
+import matplotlib.pyplot as plt
+from pysteps.exceptions import MissingOptionalDependency
+
+try:
+    import skimage
+
+    SKIMAGE_IMPORTED = True
+except ImportError:
+    SKIMAGE_IMPORTED = False
+if SKIMAGE_IMPORTED:
+    import skimage.draw as skid
+    import skimage.feature as skif
+    import skimage.measure as skime
+    import skimage.morphology as skim
+    import skimage.segmentation as skis
+try:
+    import pandas as pd
+
+    PANDAS_IMPORTED = True
+except ImportError:
+    PANDAS_IMPORTED = False
+
+
+def detection(
+    input_image,
+    minref=35,
+    maxref=48,
+    mindiff=6,
+    minsize=50,
+    minmax=41,
+    mindis=10,
+    dyn_thresh=False,
+    output_feat=False,
+    time="000000000",
+):
+    """
+    This function detects thunderstorms using a multi-threshold approach. It is
+    recommended to use a 2-D Cartesian maximum reflectivity composite, however the
+    function will process any 2-D array.
+    The thunderstorm cell detection requires both scikit-image and pandas.
+
+    Parameters
+    ----------
+    input_image : array-like
+        Array of shape (m,n) containing input image, usually maximum reflectivity in
+        dBZ with a resolution of 1 km. Nan values are ignored.
+    minref : float, optional
+        Lower threshold for object detection. Lower values will be set to NaN.
+        The default is 35 dBZ.
+    maxref : float, optional
+        Upper threshold for object detection. Higher values will be set to this value.
+        The default is 48 dBZ.
+    mindiff : float, optional
+        Minimal difference between two identified maxima within same area to split area
+        into two objects. The default is 6 dBZ.
+    minsize : float, optional
+        Minimal area for possible detected object. The default is 50 pixels.
+    minmax : float, optional
+        Minimum value of maximum in identified objects. Objects with a maximum lower
+        than this will be discarded. The default is 41 dBZ.
+    mindis : float, optional
+        Minimum distance between two maxima of identified objects. Objects with a
+        smaller distance will be merged. The default is 10 km.
+    dyn_thresh: bool, optional
+        Set to True to activate dynamic lower threshold. Restricts contours to more
```

Comment:
When the dynamic threshold is activated, does it override the minmax or minref value? 
It is not clear how it operates.

Reply:
In general it overrides the minref value, however this module has been removed for now, as I found additional issues.